core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -30,6 +30,7 @@ import scala.util.Properties
 import org.apache.commons.lang3.StringUtils
 import org.apache.hadoop.fs.Path
 import org.apache.hadoop.security.UserGroupInformation
+import org.apache.hadoop.yarn.conf.YarnConfiguration
```

Comment:
I don't think you need the explicit dependency (otherwise this would not be compiling). This is probably being brought by hadoop-client already.